src/inspect_ai/model/_providers/anthropic.py

```diff
@@ -397,6 +401,148 @@ def handle_bad_request(self, ex: BadRequestError) -> ModelOutput | Exception:
         else:
             return ex
 
+    async def resolve_chat_input(
```

Comment:
This change isn't as big as it looks. It looks bigger because I moved module level functions to class instance functions so that helper methods on `self` could be used.